products/appengine/terraform.yaml

```diff
@@ -70,6 +70,62 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           service_name: "ae-service"
         test_env_vars:
           org_id: :ORG_ID
+  FlexibleAppVersion: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: ["apps/{{project}}/services/{{service}}/versions/{{version_id}}"]
+    mutex: "apps/{{project}}"
+    error_retry_predicates: ["isAppEngineRetryableError"]
+    parameters:
+      service: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+        required: false
+    virtual_fields:
+      - !ruby/object:Provider::Terraform::VirtualFields
+        name: 'noop_on_destroy'
+        description: |
+          If set to `true`, the application version will not be deleted.
+      - !ruby/object:Provider::Terraform::VirtualFields
+        name: 'delete_service_on_destroy'
+        description: |
+          If set to `true`, the service will be deleted if it is the last version.    
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      custom_delete: templates/terraform/custom_delete/appversion_delete.go.erb
+      test_check_destroy: templates/terraform/custom_check_destroy/appengine.go.erb
+      encoder: templates/terraform/encoders/flex_app_version.go.erb
+    properties:
+      id: !ruby/object:Overrides::Terraform::PropertyOverride
+        name: 'version_id'
+      deployment: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      entrypoint: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      envVariables: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      threadsafe: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      instanceClass: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true   
```

Comment:
This feels like a lot of ignore read values. Some like `deployment` make more sense to not care about, but `instanceClass` seems like something Terraform would want to correct. Adding something to the description to indicate that upstream changes are ignored might be helpful for some of them if there is a reason that's not super intuitive. For Terraform specific behavior like this you can override `description` here in the `terraform.yaml`.